draft-schwenkschuster-oauth-spiffe-client-auth.md

```diff
@@ -279,32 +279,135 @@ Trust domain "example.org": Workload API at unix:///var/secrets/spiffe/agent.soc
+            X509v3 Basic Constraints: critical
+                CA:TRUE
+            X509v3 Key Usage: critical
+                Certificate Sign, CRL Sign
+            X509v3 Subject Alternative Name:
+                URI:spiffe://example.org
+    Signature Algorithm: ecdsa-with-SHA512
+    Signature Value:
+        30:64:02:30:0a:e9:fd:d4:cd:99:52:90:cb:14:86:93:4e:f8:
+        02:52:d6:17:12:9f:2e:65:99:0e:38:b6:b9:a6:fe:43:0f:60:
+        30:04:87:ec:24:20:80:a4:75:ee:3c:ad:9d:a2:72:0d:02:30:
+        55:93:0e:14:8c:47:47:3b:74:7c:a7:2a:2a:96:1d:a4:85:46:
+        4f:3f:95:a4:c2:ab:3c:2e:04:b3:1b:cf:02:0f:33:fc:dd:dc:
+        d5:2f:44:c8:2a:dc:ce:3f:c5:c6:89:d0
+~~~
+
+> Arndt: Bundle doesn't match X509-SVID. This needs to be fixed.
 
-The bundle endpoint is not discoverable from the JWT-SVID and X509-SVID and MUST be configured manually out of band. Bundle endpoints MUST be keyed by the trust domain identifier.
+## Alternative methods to avoid
 
-## Alternative validation methods to avoid
+The following key distribution mechanisms are alternatives and SHOULD be avoided for interopability reasons.
 
-## Manual configuration
+### SPIFFE Workload API
+
+The SPIFFE Workload API allows workloads to retrieve a trust bundle from SPIFFE. It requires the authorization server to be part of a SPIFFE trust domain and be considered a workload within it. The authors acknowledge that using the SPIFFE Workload API can reduce the time an update of the bundle is received by the authorization server.
+
+In addition to the trust bundle of the trust domain the workload resides in, the SPIFFE Workload API also allows to retrieve trust bundles from federated trust domains. This mechanism moves trust establishment away from the authorization server to the SPIFFE configuration and is NOT RECOMMENDED and more explicit configuration SHOULD be maintained.
+
+### Manual configuration
 
 In small, static environments the authorization server MAY be configured with the SPIFFE bundles manually. This approach requires human interaction to set up, rotate and manage keying material and is thus generally NOT RECOMMENDED.
 
-## Using the JWT-SVID `iss` claim
+### Using the system trust store
+
+X509-SVIDs MUST NOT be validated using the system trust store. The SPIFFE ID carried in the URI SAN is not a verified attribute in the broader X.500 ecosystem and using the system trust store as trust anchor would allow ANY certificate authority in it to issue a X509-SVID for ANY SPIFFE-ID which would be considered trusted. In comparison: using SPIFFE-native validation methods restricts the signing of SPIFFE-IDs to the corresponding trust domain signing keys.
+
+### Using the JWT-SVID `iss` claim
 
 JWT-SVIDs carrying `iss` claims could technically be validated by retrieving the signing keys via OpenID Connect Discovery or OAuth 2.0 Authorization Server Metadata. In the context of this specification these key distribution mechanisms MUST NOT be used.
```

Comment:
Add a sentence on why - e.g. the iss claim points to a SPIFFE issues and cannot normally be resolved or something similar.